src/test/java/com/softserve/teamproject/service/impl/GroupServiceImplTest.java

```diff
@@ -0,0 +1,453 @@
+    entityManager.detach(edited);
+
+    //Act
+    edited.setName("DP-116");
+    groupService.updateGroup(edited, edited.getStatus(), COORDINATOR);
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITHOUT_GROUPS)
+  @Test(expected = AccessDeniedException.class)
+  public void updateGroup_TeacherWithoutGroupEditName_ExceptionThrown() {
+    //Arrange
+    Group edited = groupRepository.findAll().get(0);
+
+    //Act
+    groupService.updateGroup(edited, edited.getStatus(), TEACHER_WITHOUT_GROUPS);
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITH_GROUPS)
+  @Test
+  public void updateGroup_TeacherWithGroupEditNewNameAndCleanExperts_EditGroup() {
+    //Arrange
+    Group edited = TestData.getGroup("EDITED-GROUP");
+    edited.setId(1);
+    edited.setTeachers(new LinkedHashSet<User>() {{
+      add(userRepository.getUserByNickName(TEACHER_WITH_GROUPS));
+    }});
+    edited.setStatus(statusRepository.getStatusByName("in-process"));
+    edited.setLocation(locationRepository.findOne(1));
+    //Act
+    GroupResource updatedResource = groupService
+        .updateGroup(edited, edited.getStatus(), TEACHER_WITH_GROUPS);
+    Group updated = groupRepository.findOne(edited.getId());
+    //Assert
+    assertEquals(edited.getName(), updated.getName());
+    assertEquals(edited.getExperts(), updated.getExperts());
+    assertEquals(edited.getName(), updatedResource.getName());
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_WITH_GROUPS)
+  @Test(expected = ValidationException.class)
+  public void updateGroup_TeacherWithGroupEditNewNameNotValidValue_ExceptionThrown() {
+    //Arrange
+    Group edited = groupRepository.findByTeachers_NickName(TEACHER_WITH_GROUPS).get(0);
+
+    //Act
+    edited.setName("EDITED_GROUP");
+    groupService.updateGroup(edited, edited.getStatus(), TEACHER_WITH_GROUPS);
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_LOCATION_WITHOUT_GROUPS)
+  @Test
+  public void getGroupResourcesFromUserLocation_TeacherLocationWithoutGroups_ReturnEmpty() {
+    //Arrange
+    final int EXPECTED_SIZE = 0;
+
+    //Act
+    Set<GroupResource> groups = groupService
+        .getGroupResourcesFromUserLocation(TEACHER_LOCATION_WITHOUT_GROUPS);
+
+    //Assert
+    assertEquals(EXPECTED_SIZE, groups.size());
+  }
+
+  @TestGroup
+  @WithUserDetails(TEACHER_LOCATION_WITH_GROUPS)
+  @Test
+  public void getGroupResourcesFromUserLocation_TeacherLocationWithGroups_ReturnGroups() {
+    //Arrange
+    final int EXPECTED_SIZE = 2;
+    List<String> expected = new ArrayList() {{
+      add("DP-115");
+      add("DP-116");
+    }};
+
+    //Act
+    Set<GroupResource> groups = groupService
+        .getGroupResourcesFromUserLocation(TEACHER_LOCATION_WITH_GROUPS);
+    int result = (int) groups.stream()
+        .map(GroupResource::getName)
+        .filter(name -> expected.contains(name))
+        .count();
+
+    //Assert
+    assertEquals(EXPECTED_SIZE, result);
+  }
```

Comment:
Overall please align all tests names to follow some single naming rule. Like ```camelCasePrepositon_camelCaseAddition_camelCaseExpectedResult```. As of now you have some variance in that:
* getGroupResourcesFromUserLocation_TeacherLocationWithGroups_ReturnGroups
* getGroupsByFilter_threeLocationOneNotExist_ReturnGroups
* getAllGroupResources_teacher_returnAllGroupResources

Also please make consistent styling within Arrange - Act - Assert blocks. I like additional blank lines between those blocks that you have in the end of the file. But they are missing in the beginning of the file